### PR TITLE
Add logging of context store

### PR DESCRIPTION
**Pull request by HiroyasuNishiyama**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

Currently, context store information is not shown at Node-RED startup and specifying unknown name as context store silently fallbacks to default store.  But, in some cases, these information are useful for solving runtime problems.
This PR add logging feature for context store.

- log at startup
```
19 Jul 09:25:46 - [info] Context store : 'memory' [module=memory]
```
- log at fallback to default store (shown first time)
```
19 Jul 09:27:46 - [warn] Unknown store 'undefined-store' specified. Use default store instead.
```

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by knolleary** [comment]
So would this mean `.info` is now a required part of the context store api?

I'm not sure I like having `logStores` as a part of the component's api. I need to look into this in more detail; I'd like to find a more natural way to log the information.

**Comment by knolleary** [comment]
An alternative approach is to move where `redNodes.loadContextsPlugin` is called.
It is currently called [very early in the loading sequence](https://github.com/node-red/node-red/blob/45e7ad8049a457e0ae4c02cddf07259aeab652e8/red/runtime/index.js#L93) - before we scan for node modules.

I think we can move the context load to [after that point](https://github.com/node-red/node-red/blob/45e7ad8049a457e0ae4c02cddf07259aeab652e8/red/runtime/index.js#L166), before we load the configured flows:

```
redNodes.loadContextsPlugin().then(function() {
    redNodes.loadFlows().then(redNodes.startFlows).catch(function(err) {});
});
```

This is the point where we already log the settings file location, so would be a natural place to log the context store details.

I'm not sure about adding `info` to the context store api. I would suggest:

 1. log the name of the store and module name (if known)
 2. if we really need to, add a `toString` function to the store and use that. We should check if a store has a custom toString function, otherwise just log the name/module as above.

**Comment by HiroyasuNishiyama** [comment]
In case we can move the context load to appropriate point, we can use implicit logging without `logStores` and `info` functions.
I will try to update current code along your suggestion.

**Comment by HiroyasuNishiyama** [comment]
Updated code not to use `logStores` and `info` function.

File: red/runtime/locales/en-US/runtime.json [hunk 1]
```diff
@@ -163,7 +163,9 @@
         "error-module-not-defined": "'module' is not defined in '__storage__' of settings.contextStorage",
         "error-invalid-module-name": "Invalid context store name: '__name__'",
         "error-invalid-default-module": "Invalid storage '__storage__' is specified as a default storage",
-        "error-use-undefined-storage": "Undefined storage '__storage__' is specified"
+        "error-use-undefined-storage": "Undefined storage '__storage__' is specified",
+        "unknown-store": "Unknown store '__name__' specified. Use default store instead.",
+        "log-store-init": "Context store  : '__name__' [__info__]"
```
Comment (knolleary) [inline red/runtime/locales/en-US/runtime.json]: Looks good - will merge.

Before you do the Japanese translations of these messages please note I want to change them slightly - there is an inconsistent use of `storage` and `store` to tidy up. Will try to get that done this weekend.

**Pull request merged**